app/src/main/java/com/github/godspeed010/weblib/ContextExtensions.kt

```diff
@@ -0,0 +1,20 @@
+package com.github.godspeed010.weblib
+
+import android.app.Activity
+import android.content.Context
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import androidx.fragment.app.Fragment
+
+fun Fragment.hideKeyboard() {
+    view?.let { activity?.hideKeyboard(it) }
+}
+
+fun Activity.hideKeyboard() {
+    hideKeyboard(currentFocus ?: View(this))
+}
```

Comment:
This function never gets called. When hideKeyboard() is called, it goes from `Fragment.hideKeyboard()` to `Context.hideKeyboard()`

I used Log statements and the Debugger to check this.

Reply:
This function was intended for if the keyboard ever needed to be hidden from an Activity, though after going through your code, I am realizing that it's not needed in this app since almost everything that handles the keyboard is in a fragment or context related. The code for the ContextExtensions.kt was based off https://stackoverflow.com/a/45857155, as I am not as familiar with Android development when it came to hiding the keyboard, and was having difficulties with the keyboard not closing when any of the color options were clicked earlier on.